src/h5web/providers/hsds/api.ts

```diff
@@ -56,7 +56,7 @@ export class HsdsApi extends ProviderApi {
         } catch {
           // https://github.com/HDFGroup/hsds/issues/87
           try {
-            return JSON.parse(data.replaceAll('NaN', '"NaN"'));
+            return JSON.parse(data.replaceAll(/(NaN|-?Infinity)/gu, '"$&"'));
```

Comment:
Extends https://github.com/silx-kit/h5web/pull/639 for `Infinity` and `-Infinity` that have the same problem as `NaN`